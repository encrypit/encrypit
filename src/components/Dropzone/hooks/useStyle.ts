import { useMemo } from 'react';

import { acceptStyle, baseStyle, focusedStyle, rejectStyle } from '../styles';

export function useStyle({
  isFocused = false,
  isDragAccept = false,
  isDragReject = false,
}) {
  return useMemo(
    () => ({
      ...baseStyle,
      ...(isFocused && focusedStyle),
      ...(isDragAccept && acceptStyle),
      ...(isDragReject && rejectStyle),
    }),
    [isFocused, isDragAccept, isDragReject],
  );
}
