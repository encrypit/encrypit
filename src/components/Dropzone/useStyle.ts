import { useMemo } from 'react';

import { acceptStyle, baseStyle, focusedStyle, rejectStyle } from './styles';

export default function useStyle({
  isFocused = false,
  isDragAccept = false,
  isDragReject = false,
}) {
  return useMemo(
    () => ({
      ...baseStyle,
      ...(isFocused && focusedStyle),
      ...(isDragAccept && acceptStyle),
      ...(isDragReject && rejectStyle),
    }),
    [isFocused, isDragAccept, isDragReject]
  );
}
