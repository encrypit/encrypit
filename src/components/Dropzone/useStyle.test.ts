import { renderHook } from '@testing-library/react';

import useStyle from './useStyle';

it('renders base style by default', () => {
  const { result } = renderHook(() => useStyle({}));
  expect(result.current).toMatchInlineSnapshot(`
    {
      "alignItems": "center",
      "backgroundColor": "#fafafa",
      "borderColor": "#eeeeee",
      "borderRadius": 2,
      "borderStyle": "dashed",
      "borderWidth": 2,
      "color": "#bdbdbd",
      "cursor": "pointer",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "outline": "none",
      "padding": "20px",
      "transition": "border .24s ease-in-out",
    }
  `);
});

it.each([
  { isFocused: false, isDragAccept: false, isDragReject: false },
  { isFocused: true, isDragAccept: false, isDragReject: false },
  { isFocused: false, isDragAccept: true, isDragReject: false },
  { isFocused: false, isDragAccept: false, isDragReject: true },
])(
  'renders style when isFocused=$isFocused, isDragAccept=$isDragAccept, isDragReject=$isDragReject',
  (props) => {
    const { result } = renderHook(() => useStyle(props));
    expect(result.current).toMatchSnapshot();
  }
);
