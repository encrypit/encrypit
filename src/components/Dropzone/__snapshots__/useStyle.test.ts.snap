// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders style when isFocused=false, isDragAccept=false, isDragReject=false 1`] = `
{
  "alignItems": "center",
  "backgroundColor": "#fafafa",
  "borderColor": "#eeeeee",
  "borderRadius": 2,
  "borderStyle": "dashed",
  "borderWidth": 2,
  "color": "#bdbdbd",
  "cursor": "pointer",
  "display": "flex",
  "flex": 1,
  "flexDirection": "column",
  "outline": "none",
  "padding": "20px",
  "transition": "border .24s ease-in-out",
}
`;

exports[`renders style when isFocused=false, isDragAccept=false, isDragReject=true 1`] = `
{
  "alignItems": "center",
  "backgroundColor": "#fafafa",
  "borderColor": "#ff1744",
  "borderRadius": 2,
  "borderStyle": "dashed",
  "borderWidth": 2,
  "color": "#bdbdbd",
  "cursor": "pointer",
  "display": "flex",
  "flex": 1,
  "flexDirection": "column",
  "outline": "none",
  "padding": "20px",
  "transition": "border .24s ease-in-out",
}
`;

exports[`renders style when isFocused=false, isDragAccept=true, isDragReject=false 1`] = `
{
  "alignItems": "center",
  "backgroundColor": "#fafafa",
  "borderColor": "#00e676",
  "borderRadius": 2,
  "borderStyle": "dashed",
  "borderWidth": 2,
  "color": "#bdbdbd",
  "cursor": "pointer",
  "display": "flex",
  "flex": 1,
  "flexDirection": "column",
  "outline": "none",
  "padding": "20px",
  "transition": "border .24s ease-in-out",
}
`;

exports[`renders style when isFocused=true, isDragAccept=false, isDragReject=false 1`] = `
{
  "alignItems": "center",
  "backgroundColor": "#fafafa",
  "borderColor": "#2196f3",
  "borderRadius": 2,
  "borderStyle": "dashed",
  "borderWidth": 2,
  "color": "#bdbdbd",
  "cursor": "pointer",
  "display": "flex",
  "flex": 1,
  "flexDirection": "column",
  "outline": "none",
  "padding": "20px",
  "transition": "border .24s ease-in-out",
}
`;
